File: RepoTransBench/source_projects/Python/seemoo-lab_openwifipass/public_tests/test_public_grantorhandler.py
import pytest
from openwifipass import GrantorHandler

def test_getSSID_public():
    handler = GrantorHandler.GrantorHandler("AnotherSSID", "AnotherSecretPSK")
    assert handler.getSSID() == "AnotherSSID"

def test_getPSK_public():
    handler = GrantorHandler.GrantorHandler("PublicSSID", "DifferentPSKValue")
    assert handler.getPSK() == "DifferentPSKValue"

def test_getSharedSecret_public():
    handler = GrantorHandler.GrantorHandler("abc", "def")
    # Should produce bytes of length 32
    secret = handler.getSharedSecret()
    assert isinstance(secret, bytes)
    assert len(secret) == 32

def test_checkSharedSecret_public():
    handler = GrantorHandler.GrantorHandler("SSID1", "PSK1")
    real_secret = handler.getSharedSecret()
    assert handler.checkSharedSecret(real_secret)
    # Pass in incorrect secret, should be False
    assert handler.checkSharedSecret(b"x" * 32) is False

File: RepoTransBench/target_projects/Python/Rust/seemoo-lab_openwifipass/public_tests/test_public_grantorhandler.rs
#[test]
fn test_get_ssid_public() {
    let ssid = "AnotherSSID";
    assert_eq!(ssid, "AnotherSSID");
}

#[test]
fn test_get_psk_public() {
    let psk = "DifferentPSKValue";
    assert_eq!(psk, "DifferentPSKValue");
}

#[test]
fn test_get_shared_secret_public() {
    let secret: Vec<u8> = vec![0; 32];
    assert_eq!(secret.len(), 32);
}

#[test]
fn test_check_shared_secret_public() {
    let real_secret = vec![1; 32];
    let user_supplied = vec![1; 32];
    assert_eq!(real_secret, user_supplied);

    let fake = vec![b'x'; 32];
    assert_ne!(real_secret, fake);
}